COMMO.GameServer/OTBParsing/OTBTreeBuilder.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+
+namespace COMMO.GameServer.OTBParsing {
+
+	public sealed class OTBTreeBuilder {
+		private readonly byte[] _serializedTreeData;
+		private readonly Stack<int> _nodeStarts = new Stack<int>();
+		private readonly Stack<OTBNodeType> _nodeTypes = new Stack<OTBNodeType>();
+		private readonly Stack<int> _childrenCounts = new Stack<int>();
+		private readonly Stack<OTBNode> _builtNodes = new Stack<OTBNode>();
+
+		public OTBTreeBuilder(byte[] serializedTreeData) {
+			if (serializedTreeData == null)
+				throw new ArgumentNullException(nameof(serializedTreeData));
+
+			_serializedTreeData = serializedTreeData;
+		}
+
+		/// <remarks>Start will be included in the data.</remarks>
+		public void AddNodeDataBegin(int start, OTBNodeType type) {
+			if (start < 0 || start > _serializedTreeData.Length)
```

Comment:
If all those conditions in the whole class didn't exist, besides the ones inside the constructor, most of the exceptions is still throwed if sent an invalid data. I recomend remove they and keep just the vital ones to avoid performance issues

Reply:
Indeed a exception will be thrown, but this way the exception is thrown as soon as the invalid data arrives, not when we try to build the three.
This makes debugging far easier.
Also, since this class is only used during the "server boot" (when parsing a .otb files), I don't think performance really matters here.
Besides, we we'll only load .otb files when the server can't find our own files.